org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporterTest.java

```diff
@@ -58,4 +65,25 @@ public void testJavaImportExclusions() throws Exception {
 			javaImportExclusions.remove(GRADLE1_PATTERN);
 		}
 	}
+
+	@Test
+	public void testGradlePersistence() throws Exception {
+		importProjects("gradle/nested");
+		List<IProject> projects = ProjectUtils.getGradleProjects();
+		for (IProject project : projects) {
+			assertTrue(GradleProjectImporter.shouldSynchronize(project.getLocation().toFile()));
+		}
+		Job.getJobManager().join(CorePlugin.GRADLE_JOB_FAMILY, new NullProgressMonitor());
+		GradleBuildSupport.saveModels();
+		for (IProject project : projects) {
+			assertFalse(GradleProjectImporter.shouldSynchronize(project.getLocation().toFile()));
+		}
+		IProject project = WorkspaceHelper.getProject("gradle1");
+		File gradleBuild = new File(project.getLocation().toFile(), "build.gradle");
+		Thread.sleep(1000);
+		gradleBuild.setLastModified(System.currentTimeMillis());
+		assertTrue(GradleProjectImporter.shouldSynchronize(project.getLocation().toFile()));
+		Job.getJobManager().join(CorePlugin.GRADLE_JOB_FAMILY, new NullProgressMonitor());
```

Comment:
why are you joining on the gradle job here? there are no other assertions

Reply:
Other tests sometimes fail when the gradle project aren't finished,